security_utils.py

```
@@ -0,0 +1,57 @@
+
+def setup_global_logging_redaction():
+    """
+    Attach the SecretRedactorFilter to all handlers of the root logger.
+    """
+    redactor = SecretRedactorFilter()
+    root_logger = logging.getLogger()
+    for handler in root_logger.handlers:
+        handler.addFilter(redactor)
+    # Also add to any loggers that are created with logging.getLogger(__name__)
+    # (Optional: recursively add to all known loggers)
+    for logger_name in logging.root.manager.loggerDict:
```

Pylint (3.3.7)

[error] Instance of 'RootLogger' has no 'loggerDict' member

(E1101)